fix: schema changes for couterparty

liorzam opened the pull request:
## **Type**
enhancement


___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>migration.sql</strong><dd><code>Simplify TransactionRecord and Alert Tables by Removing Unused Columns</code></dd></summary>
<hr>

services/workflows-service/prisma/migrations/20240305210407_transaction_monitoring/migration.sql
<li>Removed <code>endUserId</code> and <code>businessId</code> columns from <code>TransactionRecord</code> and <br><code>Alert</code> tables.<br> <li> Removed indexes and foreign key constraints related to <code>endUserId</code> and <br><code>businessId</code>.


</details>
    

  </td>
  <td><a href="https://github.com/ballerine-io/ballerine/pull/2160/files#diff-d09030b0d66144a37a08f36e88021ba39d6874505654c85440cef0128812a873">+0/-22</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

MatanYadaev commented on services/workflows-service/prisma/migrations/20240305210407_transaction_monitoring/migration.sql:
```diff
@@ -126,8 +126,6 @@ CREATE TABLE "TransactionRecord" (
     "projectId" TEXT NOT NULL,
     "counterpartyOriginatorId" TEXT,
     "counterpartyBeneficiaryId" TEXT,
-    "endUserId" TEXT,
```
Don't you prefer creating a new migration?
It's still in an epic branch, so you did quite ok, but when multiple people are involved I usually prefer creating new migrations.

MatanYadaev replied:
If it's ok, I prefer you to create a new migration instead of adjusting existing one. Now I based on `epic/bal-1521-tm` and I write new migrations. If you will update existing migrations, my migrations will fail, and I'll have to regenerate them.
For the sake of all of us, let's keep migrations immutable on shared branches.

liorzam replied:
ok later we gonna remove them all and recreate a new one

liorzblrn committed:
chore(opencommit.yml): add OpenCommit Action workflow to automate commit message generation

chore(backoffice-v2): update dependencies in package.json

chore(backoffice-v2): update translations in toast.json

refactor(backoffice-v2): change alert severity, status, type, and state values to lowercase for consistency

refactor(backoffice-v2): change transaction direction and payment method values to lowercase for consistency

fix(CallToAction.tsx): change motionProps variable name to motionButtonProps for clarity and consistency
fix(CallToAction.tsx): update animate prop of MotionButton to include opacity based on value.props.disabled
fix(CallToActionLegacy.tsx): change motionProps variable name to motionButtonProps for clarity and consistency
fix(CallToActionLegacy.tsx): update animate prop of MotionButton to include opacity based on disabled prop
fix(CaseCallToActionLegacy.tsx): change motionProps variable name to motionButtonProps for clarity and consistency
fix(CaseCallToActionLegacy.tsx): update animate prop of MotionButton to include opacity based on data.disabled or isDisabled
fix(useKycBlock.tsx): change motionProps variable name to motionBadgeProps for clarity and consistency
fix(useKycBlock.tsx): update animate prop of MotionButton to include opacity based on isDisabled
fix(useAssociatedCompaniesBlock.tsx): change motionButtonProps variable name to motionButtonProps for clarity and consistency

chore(server.ts): change port variable case from lowercase port to uppercase PORT to improve semantics

feat(server.ts): add support for process.env.PORT environment variable to be able to run app on a configurable port

fix(useDirectorsBlocks.tsx): update opacity of MotionButton when isApproveDisabled is true

fix(useDocumentBlocks.tsx): update opacity of MotionButton when canApprove is false

fix(motion-badge-props.ts): change type of motionBadgeProps to match ComponentProps<typeof MotionBadge>

fix(useDefaultBlocksLogic.tsx): update opacity of MotionButton when caseState.actionButtonsEnabled is false

fix(useKybExampleBlocksLogic.tsx): update opacity of MotionButton when caseState.actionButtonsEnabled is false

fix(columns.tsx): convert status to title case in AlertsTable column

fix(get-alerts-search-schema.ts): add alertType field to search schema

fix(TransactionMonitoringAlertsAnalysis.page.tsx): add key prop to AlertAnalysisSheet to force re-render when alertId changes

fix(useTransactionMonitoringAlertsAnalysisPageLogic.tsx): navigate back twice when alertId changes

chore(CHANGELOG.md): update dependencies in kyb-app to latest versions

chore(package.json): bump version from 0.1.42 to 0.1.44 in @ballerine/kyb-app package to reflect dependency updates

chore(package.json): update dependencies in @ballerine/kyb-app package:
- @ballerine/blocks from 0.1.26 to 0.1.27
- @ballerine/common from ^0.7.41 to ^0.7.43
- @ballerine/ui from 0.3.28 to 0.3.29
- @ballerine/workflow-browser-sdk from 0.5.41 to 0.5.43

fix(JSONFormArrayFieldLayout.tsx): add missing import statement for useUIElementHandlers hook in JSONFormArrayFieldLayout component

feat(JSONFormArrayFieldLayout.tsx): add support for onChangeHandler in JSONFormArrayFieldLayout component to handle changes in array field values

fix(SubmitButton.tsx): add condition to trackFinish() function call only when the current page is the last page

chore(CHANGELOG.md): bump version from 0.1.12 to 0.1.13 in @ballerine/workflows-dashboard package

chore(package.json): bump version from 0.1.12 to 0.1.13 in @ballerine/workflows-dashboard package to reflect dependency updates

chore(package.json): update dependencies in @ballerine/workflows-dashboard package:
- @ballerine/config from ^1.0.5 to ^1.0.6
- @ballerine/eslint-config-react from ^1.0.6 to ^1.0.7

chore(CHANGELOG.md): bump version from 0.1.41 to 0.1.43 in @ballerine/headless-example package

chore(package.json): bump version from 0.1.41 to 0.1.43 in @ballerine/headless-example package to reflect dependency updates

chore(package.json): update dependencies in @ballerine/headless-example package:
- @ballerine/common from 0.7.41 to 0.7.43
- @ballerine/workflow-browser-sdk from 0.5.41 to 0.5.43

chore(CHANGELOG.md): bump version from 0.0.2 to

chore(blocks): bump version to 0.1.27 and update dependencies
chore(common): bump version to 0.7.43 and update dependencies
fix(common): update default context schema to include new document categories
fix(common): update GH document schema to use non-empty strings for account holder name and registration number

chore(UG.ts): improve variable names and types for better readability and semantics
chore(CHANGELOG.md): bump version to 1.0.6 in @ballerine/config
chore(package.json): bump version to 1.0.6 in @ballerine/config
chore(CHANGELOG.md): bump version to 1.0.7 in @ballerine/eslint-config-react
chore(package.json): bump version to 1.0.7 in @ballerine/eslint-config-react
chore(CHANGELOG.md): bump version to 1.0.6 in @ballerine/eslint-config
chore(package.json): bump version to 1.0.6 in @ballerine/eslint-config
chore(CHANGELOG.md): bump version to 1.0.14 in @ballerine/react-pdf-toolkit
chore(package.json): bump version to 1.0.14 in @ballerine/react-pdf-toolkit
chore(CHANGELOG.md): bump version to 0.4.26 in @ballerine/rules-engine-lib
chore(package.json): bump version to 0.4.26 in @ballerine/rules-engine-lib

chore(ui): bump version to 0.3.29 and update dependencies
chore(workflow-core): bump version to 0.5.43 and update dependencies
fix(server.ts): change port variable case from lowercase port to uppercase PORT to improve semantics
feat(server.ts): add support for process.env.PORT environment variable to be able to run app on a configurable port
fix(ui): remove duplicate import statement in ArrayFieldsLayoutItem.tsx
fix(workflow-core): handle error in IterativePlugin when iterate on param is not an array

chore(pnpm-lock.yaml): update dependencies in pnpm-lock.yaml

- Update '@ballerine/blocks' to version 0.1.27
- Update '@ballerine/common' to version 0.7.43
- Update '@ballerine/ui' to version ^0.3.29
- Update '@ballerine/workflow-browser-sdk' to version 0.5.43
- Update '@ballerine/workflow-node-sdk' to version 0.5.43
- Update '@ballerine/config' to version ^1.0.6
- Update '@ballerine/eslint-config-react' to version ^1.0.7
- Update '@fontsource/inter' to version ^4.5.15
- Update '@lukemorales/query-key-factory' to version ^1.0.3
- Update '@cspell/cspell-types' to version ^6.31.1
- Update '@babel/core' to version 7.17.9
- Update '@rollup/plugin-babel' to version 5.3.1
- Update 'eslint-plugin-react' to version ^7.33.2
- Update 'react' to version ^18.2.0
- Update 'react-dom' to version ^18.2.0
- Update 'react-router-dom' to version ^6.2.1
- Update 'react-scripts' to version 5.0.0
- Update 'react-test-renderer' to version ^18.2.0
- Update 'react-pdf' to version ^2.0.17
- Update 'react-pdf/renderer' to version ^3.1.14
- Update 'react-query' to version ^3.22.4
- Update 'react-query-devtools' to version ^3.22.4
- Update 'react-query/react' to version ^3.22.4
- Update 'react-query/types' to version ^3.22.4
- Update 'react-query/types/react' to version ^3.22.4
- Update 'react-query/types/react-query' to version ^3.22.4
- Update 'react-query/types/react-query/react' to version ^3.22.4
- Update 'react-query/types/react-query/react-query' to version ^3.22.4
- Update 'react

chore(pnpm-lock.yaml): update versions of '@ballerine/config', '@ballerine/eslint-config', '@ballerine/common', '@ballerine/workflow-core', and '@ballerine/workflow-node-sdk' to '^1.0.6', '^1.0.6', '0.7.43', '0.5.43', and '0.5.43' respectively
feat(pnpm-lock.yaml): add new dev dependency 'eslint-plugin-ballerine' with version '1.0.0'

chore(web-ui-sdk): add .npmignore file to exclude node_modules directory from npm package

chore(web-ui-sdk): update dependencies in package.json
- @ballerine/common@0.7.43

chore(workflow-browser-sdk): update dependencies in package.json
- @ballerine/common@0.7.43
- @ballerine/workflow-core@0.5.43

chore(workflow-node-sdk): update dependencies in package.json
- @ballerine/workflow-core@0.5.43

chore(workflows-service): update dependencies in package.json
- @ballerine/common@0.7.43
- @ballerine/workflow-core@0.5.43
- @ballerine/workflow-node-sdk@0.5.43

chore(workflows-service): update subproject commit hash in data-migrations
chore(workflows-service): remove unused migration files
chore(workflows-service): update transaction monitoring migration.sql with new enum values and column changes

chore(workflows-service): add migration to fix transaction monitoring relations

This commit adds a new migration file to fix the transaction monitoring relations in the workflows-service. The migration includes the following changes:

- Drops the columns `businessId` and `endUserId` from the `Alert` table, resulting in the loss of all data in these columns.
- Drops the columns `businessId` and `endUserId` from the `TransactionRecord` table, resulting in the loss of all data in these columns.

Please note that this migration will also drop the foreign key constraints and indexes associated with these columns.

chore(schema.prisma): remove unused fields and update enum values for TransactionRecordType, ReviewStatus, VerificationStatus, TransactionRecordStatus, PaymentMethod, PaymentType, PaymentChannel, PaymentIssuer, PaymentGateway, PaymentAcquirer, PaymentProcessor, PaymentBrandName, TransactionDirection, AlertSeverity, AlertState, AlertStatus, AlertType, CounterpartyType, RiskCategory, ComplianceStatus, PEPStatus, and SanctionListMatchStatus for better consistency and readability

fix(generate-alerts.ts): import correct types from the data-analytics/types file
fix(generate-alerts.ts): change rule properties from snake_case to camelCase for consistency
fix(generate-alerts.ts): change payment method values from uppercase to lowercase for consistency
fix(generate-alerts.ts): change direction values from uppercase to lowercase for consistency
fix(generate-alerts.ts): change timeUnit value from 'days' to 'day' for consistency
fix(generate-alerts.ts): change amountThreshold value from 5 to 1000 for consistency
fix(generate-alerts.ts): change amountBetween value from { min: 500, max: 1000 } to { min: 500, max: 999 } for consistency
fix(generate-alerts.ts): change ruleId values from numbers to string values for consistency
fix(generate-alerts.ts): change rulesetId values from numbers to string values for consistency
fix(generate-alerts.ts): change cardBIN property to cardBin for consistency
fix(generate-transactions.ts): change cardBIN property to cardBin for consistency

fix(seed.ts): rename generateFakeAlertDefinition to generateFakeAlertDefinitions for clarity and consistency
fix(seed.ts): remove unused import for generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused import for Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateKycManualReviewRuntimeAndToken
fix(seed.ts): remove unused variable Type
fix(seed.ts): remove unused variable BCRYPT_SALT
fix(seed.ts): remove unused variable generateTransactions
fix(seed.ts): remove unused variable generateK

refactor(alert): remove unused imports and parameters in AlertControllerInternal.checkAlerts method

refactor(alert): remove unused imports in AlertModule

refactor(alert): remove unused imports and parameters in AlertRepository.create method

feat(alert): add exists method to AlertRepository to check if an alert exists

fix(alert.service.ts): import missing dependencies and remove unused dependencies
feat(alert.service.ts): add support for checking alerts based on alert definitions and inline rules
feat(alert.service.ts): add support for creating alerts based on alert definitions and subject results
feat(alert.service.ts): add support for checking duplicate alerts based on alert definitions and data
feat(alert.service.ts): add support for mapping alert states to alert statuses
feat(alert/consts.ts): add constants for alert execution statuses
feat(alert/types.ts): add type for alert response
fix(data-analytics.module.ts): remove unused DataAnalyticsRepository dependency

chore(data-analytics.repository): remove unused integration test file evaluate-transactions-against-dynamic-rules.intg.test.ts

chore(data-analytics.repository): remove unused evaluate-transactions-against-dynamic-rules.ts and index.ts files

The evaluate-transactions-against-dynamic-rules.ts and index.ts files were no longer being used and were deleted to clean up the codebase.

feat(data-analytics.service.intg.test.ts): add integration tests for the DataAnalyticsService

feat(data-analytics.service.ts): add support for running inline rules in the DataAnalyticsService

feat(data-analytics.service.ts): add evaluateTransactionsAgainstDynamicRules method to evaluate transactions against dynamic rules

fix(data-analytics.service.ts): fix conditions variable shadowing issue in evaluateTransactionsAgainstDynamicRules method

feat(data-analytics.service.ts): add support for different time units in getIntervalTime method

feat(data-analytics.service.ts): add logging for executing query and results in evaluateTransactionsAgainstDynamicRules method

feat(data-analytics.service.ts): add extendedClient property to PrismaService for extending PrismaClient functionality

feat(data-analytics.service.ts): add _evaluateNameToFunction property to DataAnalyticsService for mapping rule names to evaluation functions

feat(data-analytics.service.ts): add runInlineRule method to DataAnalyticsService for running inline rules

feat(data-analytics.service.ts): add error handling for missing evaluation function in runInlineRule method

feat(data-analytics.service.ts): add projectId parameter to evaluateTransactionsAgainstDynamicRules method

feat(types.ts): add projectId property to TransactionsAgainstDynamicRulesType

feat(prisma.service.ts): add prismaExtendedClient function for extending PrismaClient functionality

feat(prisma.service.ts): add extendedClient property to PrismaService for accessing extended PrismaClient functionality

feat(translation.json): add translation for "relationship" key in Chinese locale

feat(translation.json): add translation for "companies" key in Chinese locale

chore(translation.json): add "relationship" field as required in the translation file to improve validation message
chore(translation.json): add "companies" field as required in the translation file to improve validation message
refactor(get-transactions.dto.ts): remove unused "businessId" property from GetTransactionsDto
refactor(transaction-create.dto.ts): change "cardBIN" property to "cardBin" in TransactionCreateDto to match naming convention
refactor(transaction.controller.external.ts): remove unused "businessId" query parameter from TransactionControllerExternal
refactor(transaction.mapper.ts): change "cardBIN" property to "cardBin" in TransactionEntityMapper to match naming convention
refactor(transaction.repository.ts): remove unused "businessId" and "endUserId" variables in TransactionRepository
refactor(transaction.repository.ts): remove commented out code in TransactionRepository
refactor(transaction.repository.ts): remove unused "businessId" condition in TransactionRepository
refactor(types.ts): add type aliases for AnyArray, AnyRecord, GenericFunction, and GenericAsyncFunction in types.ts

fix(hook-callback-handler.service.ts): import missing dependencies and fix type assertion errors
feat(hook-callback-handler.service.ts): add support for updating process status in workflow context
refactor(hook-callback-handler.service.ts): improve code readability by extracting helper function
refactor(hook-callback-handler.service.ts): improve code readability by renaming variables and adding type assertions
refactor(hook-callback-handler.service.ts): improve code readability by removing unnecessary comments
fix(add-properties-schema-to-document.ts): fix variable assignment for propertiesSchema
chore(package.json): update dependencies for @ballerine/common, @ballerine/config, and @ballerine/eslint-config
